sqlglot/dataframe/sql/column.py

```diff
@@ -0,0 +1,320 @@
+from __future__ import annotations
+
+import datetime
+import typing as t
+from collections.abc import Iterable
+
+import sqlglot
+from sqlglot import expressions as exp
+from sqlglot.dataframe.sql.types import DataType
+from sqlglot.helper import flatten
+
+if t.TYPE_CHECKING:
+    from sqlglot.dataframe.sql._typing import (
+        ColumnOrName,
+        ColumnOrPrimitive,
+        DateTimeLiteral,
+        DecimalLiteral,
+        Literals,
+        Primitives,
+    )
+    from sqlglot.dataframe.sql.window import WindowSpec
+
+
+class Column:
+    def __init__(self, expression: t.Union[t.Any, exp.Expression]):
+        if isinstance(expression, str):
+            expression = sqlglot.parse_one(expression, read="spark")
+        elif isinstance(expression, Column):
+            expression = expression.expression
+        self.expression = expression
+        if self.expression is None or isinstance(self.expression, (int, float, bool, dict, Iterable, datetime.date)):
```

Comment:
would it be better to do like

if not isinstance(expression, exp.Expression):

Reply:
Yeah good call updated